File: C/command_line_arguments/command_line_arguments.c
#include <stdio.h>

int main(int argc, char *argv[])
{
    // argc always >= 1 as program name is counted

    printf("Number of arguments passed: %d\n\n", argc);

    int i;
    for (i = 0; i < argc; i++) {
        printf("Argument %d: %s\n", (i + 1), argv[i]);
    }

    printf("\nFirst argument provided: %s\n", argv[0]);
    printf("Final argument provided: %s\n", argv[argc - 1]);

    return 0;
}

File: Python/command_line_arguments/command_line_arguments.py
import sys

# len(sys.argv) always >= 1 as program name is counted

print("Number of arguments passed:", len(sys.argv))
print("List of arguments:", sys.argv)

for i in range(len(sys.argv)):
    print(f"Argument {i}: {sys.argv[i]}")

print("\nFirst argument provided:", sys.argv[0])
print("Final argument provided:", sys.argv[len(sys.argv) - 1])
